### PR TITLE
Add package-composite-v4

### PR DESCRIPTION
**Pull request by bradrich-msft**
### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/sql/resource-manager/readme.md [hunk 1]
```diff
@@ -24,13 +24,115 @@ These are the global settings for the Sql API.
...
+
+APIs must only be added to this section when the API is publicly available in at least 1 production region and at least 1 generated client has been tested end-to-end.
+
+Differences in v4 (compared to v3):
+
+* Added new API for databases and elastic pools
+
+  * `-2020-08-01/databases.json`
+
+  * `-2020-08-01/elasticPools.json`
```
Comment (hotsnr) [inline specification/sql/resource-manager/readme.md:48]: Could you please add that failover API for elastic pool was integrated into elasticPools? (if you think this is reasonable)

**Review by hotsnr — APPROVED** [review]

**Comment by markcowl** [comment]
@bradrich-msft  It looks like there are breaking change when swtiching the default tag from v3 to v4 (as shouwn in the pythion client breaking changes in the comments above).  Is this expected?  How do we plan to keep this from breaking customers?

**Comment by bradrich-msft** [comment]
> @bradrich-msft It looks like there are breaking change when swtiching the default tag from v3 to v4 (as shouwn in the pythion client breaking changes in the comments above). Is this expected? How do we plan to keep this from breaking customers?

Hi @markcowl it's not expected that bumping default tag version would break python client.  However, I admit to limited knowledge of the python client or a documented process of default tag change to avoid it.  Do you have any info to share on that?  

We are in process of also updating .NET SDK to consume this change which will be followed by powershell and CLI clients.  Is that sufficient?

**Comment by bradrich-msft** [comment]
I took a look at azure-sdk-for-python breaking changes logs and these changes are correct and expected.  In fact, we introduced v4 package for the very reason of these breaking changes.  If this is not the correct way to introduce breaking changes for python client, what is?

**Comment by bradrich-msft** [comment]
@markcowl one more thing.  We introduced a new API version, 2020-08-01-preview, which is the only API version with the breaking change.  Versions prior to 20200801 will continue to have old functionality and existing clients will work.  I have the change ready which updates .NET SDK to a new version consuming this change.  This was our plan to deal with the breaking change.  New API version and new versions for composite, SDK, clients.

A related change that has already been merged: https://github.com/Azure/azure-rest-api-specs/pull/10432

**Comment by markcowl** [comment]
@bradrich-msft Generally, breaking changes are heavily discouraged / not allowed, we will need to get the sdk team to sign off on the breaking changes

**Review by markcowl — APPROVED** [review]

**Pull request merged**